src/server.ts

```diff
@@ -107,7 +107,7 @@ class Server {
             formatError: undefined,
             callback: undefined,
           };
-          params = this.onSubscribe(parsedMessage, params);
+          params = this.onSubscribe(parsedMessage, params) || params;
```

Comment:
IMO this code is not very readable, it would be better as an if statement with an intermediate variable.